### DISCUSSION TITLE
Possible to pin a given task to a specific scheduler

### DISCUSSION THREAD

**Discussion by djarnis73** [kagkarlsson/db-scheduler#444] [Ideas]
Hi

I'm working on a project where we have many different services using a shared database. Each service has its own set of tasks (that only this service should execute) and a given service could be clustered (as in having multiple active schedulers).

I'm looking for a way to ensure that a task is executed by a particular scheduler for instance based on the scheduler name, but a more flexible way would be preferred, like perhaps a tag assigned to a task and then each scheduler could define the tags they should handle.

I could of course create multiple task tables, but that gets ugly quickly.

Would that be difficult to implement?

**Comment by djarnis73** [comment]
Ahh, I thought I had searched for this, but now I just saw https://github.com/kagkarlsson/db-scheduler/issues/442 which I guess is the same thing I'm looking for.

**Reply by kagkarlsson** [reply]
Yeah, currently no plans for this. It is simpler to have dedicated tables. Possibly somewhere down the line.. but not currently planning to support this :)

**Comment by djarnis73** [comment]
Just had an idea that you could perhaps use table partitioning (add an extra `pin` column that you partition on) and then use a `JdbcCustomization` to include that column in the query.

Then it should be possible to do the pinning without performance penalty (but at the cost of extra complexity).

But I'm not really sure how much you can alter the query with the `JdbcCustomization` this is more or less random thoughts but could perhaps be useful for others.
